File: Python/refer-offer/fan_zhuan_lian_biao_lcof.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

def reverseList(head: ListNode) -> ListNode:
    if head is None or head.next is None:
        return head
    
    front, back = head, head.next
    front.next = None

    while back.next:
        temp = back.next
        back.next = front
        front = back
        back = temp
    
    back.next = front
    return back

File: src/tmp/Java/src/com/dbc/algorithm/ReferOffer/FanZhuanLianBiaoLcof.java
package com.dbc.algorithm.ReferOffer;

public class FanZhuanLianBiaoLcof {
    public class ListNode {
        int val;
        ListNode next;

        ListNode(int x) {
            val = x;
        }
    }

    public ListNode reverseList(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }
        ListNode front = head, back = head.next;
        front.next = null;
        while (back.next != null) {
            ListNode temp = back.next;
            back.next = front;
            front = back;
            back = temp;
        }
        back.next = front;
        return back;
    }
}
